test/js/weak-test.js

```diff
@@ -21,11 +21,16 @@ test('weak', function (t) {
     count++;
   });
 
-  // run weak callback, should dispose
-  gc();
+  var timeout = setTimeout(function () {
```

Comment:
I think just wrapping it in a `setImmediate()` should be enough, no need to fuss around with clearing timers again.